src/main/java/com/mraof/minestuck/world/GateHandler.java

```diff
@@ -92,44 +90,60 @@ private static GlobalPos findPosNearLandGate(ServerLevel level)
 	
 	private static GlobalPos findClientLandGate(ServerLevel level)
 	{
-		SburbConnection landConnection = SburbHandler.getConnectionForDimension(level.getServer(), level.dimension());
-		if(landConnection != null)
+		Optional<PlayerIdentifier> landPlayer = SburbPlayerData.getForLand(level).map(SburbPlayerData::playerId);
+		if(landPlayer.isEmpty())
 		{
-			SburbConnection clientConnection = SkaianetHandler.get(level.getServer()).getPrimaryConnection(landConnection.getClientIdentifier(), false).orElse(null);
-			
-			if(clientConnection != null && clientConnection.hasEntered() && MSDimensions.isLandDimension(level.getServer(), clientConnection.getClientDimension()))
-			{
-				ResourceKey<Level> clientDim = clientConnection.getClientDimension();
-				ServerLevel clientLevel = level.getServer().getLevel(clientDim);
-				BlockPos gatePos = Type.LAND_GATE.getPosition(clientLevel);
-				if(gatePos == null)
-				{LOGGER.error("Unexpected error: Can't initialize land gate placement for dimension {}!", clientDim); return null;}
-				
-				return GlobalPos.of(clientDim, gatePos);
-			}
-			//else player.sendMessage(new TranslationTextComponent(MISSING_LAND));
-		} else
-			LOGGER.error("Unexpected error: Can't find connection for dimension {}!", level.dimension());
-		return null;
+			LOGGER.error("Unexpected error: Can't find player for land {}!", level.dimension());
+			return null;
+		}
+		
+		Optional<ResourceKey<Level>> clientLandOptional = SburbConnections.get(level.getServer())
+				.primaryPartnerForServer(landPlayer.get())
+				.flatMap(clientPlayer -> {
+					SburbPlayerData clientPlayerData = SburbPlayerData.get(clientPlayer, level.getServer());
+					return Optional.ofNullable(clientPlayerData.getLandDimensionIfEntered());
+				});
+		
+		if(clientLandOptional.isEmpty())
+		{
+			//player.sendMessage(new TranslationTextComponent(MISSING_LAND));
+			return null;
+		}
```

Comment:
Is the MISSING_LAND message intended to eventually be used? Is there another reason it is being kept around?

Reply:
I had left it untouched to avoid deciding anything on it and focus more on other changes.
Looking at it now, it seems that it was originally commented out due to a code refactoring that left this code in a function without a player parameter, and I didn't think it was appropriate to add it there.
I'm not sure how much value the message holds, so I'll go ahead and remove the commented messages and the `MISSING_LAND` key.